### PR TITLE
add definition file

### PR DESCRIPTION
**Pull request by lostfictions**

### CODE REVIEW

File: blake2s.d.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+export as namespace BLAKE2s;
+
+export type ByteArray = number[] | Uint8Array;
+
+export default class BLAKE2s {
+  constructor(digestLength: number | undefined, keyOrConfig: ByteArray | BLAKE2sConfig);
```
Comment (dchest) [inline blake2s.d.ts]: `keyOrConfig` is optional.

Can a type definition contain multiple declarations for function that accepts different arguments, like normal TypeScript?

```
constructor(digestLength: number?, key?: ByteArray);
constructor(digestLength: number?, config?: BLAKE2sConfig);
```

I wonder if it also can specify the default value as `digestLength = 32`?

File: blake2s.d.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  processBlock(length: number): void;
```
Comment (dchest) [inline blake2s.d.ts]: This is actually a private method.

File: blake2s.d.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  update(p: ByteArray, offset?: number, length?: number): this;
+  digest(): Uint8Array;
+  hexDigest(): string;
+  digestLength: number;
+  blockLength: number;
+  keyLength: number;
+  saltLength: number;
+  personalizationLength: number;
```
Comment (dchest) [inline blake2s.d.ts]: These numbers are `readonly`.

Comment (lostfictions) [reply blake2s.d.ts]: Oops, missed that! I'm honestly not sure how to resolve the incompatibility between ES modules and UMD defs, so I've declared them as both `static readonly` members and const fields in the namespace -- that seems least likely to break something for a consumer?

File: blake2s.d.ts (hunk 1 above)
Comment (lostfictions) [reply blake2s.d.ts]: Fixed with an overloaded constructor. You can't provide a parameter initializer in defs, unfortunately.

**Comment by lostfictions** [comment]
Fixed, hopefully. I also added a `tsconfig.json` so the definition file would be linted, and to check the defs against the tests file. That said, I didn't bother to fix or suppress any errors in the tests file other than those related to the defs. Hope that helps!

**Pull request merged**